lib/parser.js

```diff
@@ -117,14 +117,23 @@ function parseSchema( value, engine ) {
         delete value.type;
     }
 
-    let parser = types[ type ];
+    let schemaParser = null;
 
-    if( !parser ) {
+    // Predefined type.
+    if (types[type]) {
+        schemaParser = types[type];
+    }
+    // Custom type - use Base schema parser.
```

Comment:
I will update the comment here to better define what the check for `utils.isFunction(engine[type])` does/signifies.

Ultimately, types are just functions on the base engine (joi). The original lib has some special parsing for some joi types, etc, but fore the most part, the type and its options are pretty much all functions, and that's what the parser does.

For adding a custom type, by checking that the type exists as a function, is good enough to use it with no additional functionality changes, and let the parser process the config as per any other normal/basic type.